app/models/weam.rb

```diff
@@ -40,15 +46,15 @@ class Weam < ActiveRecord::Base
     'ope' => { column: :ope, converter: OpeConverter }
   }.freeze
 
-  validates :facility_code, :institution, :institution_type, presence: true
+  validates :facility_code, :institution, :institution_type_name, presence: true
   validates :bah, numericality: true, allow_blank: true
 
   before_validation :derive_dependent_columns
 
   # Computes all fields that are dependent on other fields. Called in validation because
   # activerecord-import does not engage callbacks when saving
   def derive_dependent_columns
-    self.institution_type = derive_type
+    self.institution_type_name = derive_type
```

Comment:
Not suggesting over-riding #initialize. It looks like you're using `before_validation` right now, I'm not sure if that is better than using `after_initialize.` What if I want to manually override an attribute before saving? I might not be able to because before_validation would possibly change my manual override. 

Just something to think about. In any case, I think making `derive_dependent_columns` a private method makes sense.